### PR TITLE
User lectures bug fix

### PR DESCRIPTION
**Pull request by AriLFrankel**
user lectures bug fix (validate they're not already in that table for that lecture before inserting them)
pass complete lecture profile to audience member on join so that they get the same lecture state as the presenter (and everyone else) has

### CODE REVIEW

File: client/src/components/JoinPresBox.jsx [hunk 1]
```diff
@@ -79,11 +79,43 @@ class JoinPresBox extends Component {
         role: 'audience'
       };
       socket.emit('userLecture', lecture);
+      console.log('questions: ', questions);
+      // Dispatch all of the questions and displayed boolean into the store
+      Object.keys(questions).slice(1).forEach((questionId) => {
```
Comment (rtopol2) [inline client/src/components/JoinPresBox.jsx]: Not necessarily removing correct key

File: client/src/reducers/questionReducer.jsx [hunk 2]
```diff
@@ -12,7 +12,7 @@
 // action = {type: 'TOGGLE_UPVOTED', questionId: string}
 // action = {type: 'CLEAR_QUESTIONS'}
 
-const questionReducer = (state = {}, action) => {
+const questionReducer = (state = {enabled: false}, action) => {
   switch (action.type) {
     case 'CREATE_QUESTION':
       return Object.assign({}, state, {
```
Comment (rtopol2) [inline client/src/reducers/questionReducer.jsx]: need to use `action.votes || 1` to assign votes

File: client/src/components/QuestionBox.jsx [hunk 3]
```diff
@@ -76,15 +78,15 @@ class QuestionBox extends Component {
...
+    let questions = store.getState().questions;
+    let displayQuestions = questions.enabled ? 'block' : 'none';
     // Assign an id to the main component div so that it can be targeted on toggle events
     if (this.props.role === 'presenter') {
       return (
         <div id="QuestionBox" style={{display: 'none'}}>
           <input key={1} type="text" id="questionInput"></input>
           <button key={2} id="submitQuestion" onClick={this.submitQuestion.bind(this)}>Submit</button>
-          {Object.keys(questions).sort(function (a, b) {
+          {Object.keys(questions).slice(1).sort(function (a, b) {
```
Comment (rtopol2) [inline client/src/components/QuestionBox.jsx]: need to refactor to delete proper key

File: client/src/components/QuestionBox.jsx [hunk 4]
```diff
@@ -94,10 +96,10 @@ class QuestionBox extends Component {
       );
     } else {
       return (
-        <div id="QuestionBox" style={{display: 'none'}}>
+        <div id="QuestionBox" style={{display: displayQuestions}}>
           <input key={1} type="text" id="questionInput"></input>
           <button key={2} id="submitQuestion" onClick={this.submitQuestion.bind(this)}>Submit</button>
-          {Object.keys(questions).map((questionId, i) =>
+          {Object.keys(questions).slice(1).map((questionId, i) =>
```
Comment (rtopol2) [inline client/src/components/QuestionBox.jsx]: refactor to delete proper key

File: client/src/components/TitleBar.jsx [hunk 5]
```diff
@@ -6,7 +6,7 @@
 
 import React, { Component } from 'react';
 import { connect } from 'react-redux';
-
+import store from '../store.jsx';
```
Comment (rtopol2) [inline client/src/components/TitleBar.jsx]: not used. should be taken out

**Comment by JTMcDuff** [comment]
Updated with Ross and Ari.  Code now passes.  Merging.

**Pull request merged**